pkgs/top-level/all-packages.nix

```diff
@@ -7242,6 +7244,37 @@ with pkgs;
 
   freeimage = callPackage ../development/libraries/freeimage { };
 
+  freeipaKerberos = krb5Full.override { inherit libverto; };
+
+  freeipaKerberosLib = freeipaKerberos.override { type = "lib"; };
+
+  freeipaCurl = curl.override {
+    gssSupport = true;
+    gss = krb5Full;
+  };
+
+  freeipaBind = bind.override { kerberos = freeipaKerberosLib; };
+
+  freeipaSamba = samba4.override {
+    enableLDAP = true;
+    kerberos = freeipaKerberosLib;
+  };
+          
+  freeipa = callPackage ../os-specific/linux/freeipa {
+    kerberos = freeipaKerberos;
+    sasl = cyrus_sasl;
+    # ipa-join requires curl with krb5 GSSAPI delegation
+    curl = freeipaCurl;
+    inherit (python27Packages)
```

Comment:
while we typically pass in packages individually, in the case of Python we always pass in the whole set. Therefore, pass in either `python2` (and use the pkgs with `python2.pkgs`) or `python2Packages` (and get the interpreter with `python2Packages.python`).